### PR TITLE
[docs] display supersession info

### PR DESCRIPTION
**Pull request by danielgafni**
## Summary & Motivation

Adding new `Supersession` helpers for docs.

This is done in alignment with the new [API lifecycle](https://www.notion.so/dagster/API-Lifecycle-Refresh-f72f6e199da3409c8c5da297a109b3c8) and as requested in [this](https://github.com/dagster-io/dagster/pull/25335#discussion_r1805034325) comment by @yuhan 

Example screenshot in the upstack [PR](https://github.com/dagster-io/dagster/pull/25335).

### CODE REVIEW

**Comment by danielgafni** [comment]
* **#25362** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25362?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @danielgafni and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/markdoc-component-documentation.md [hunk 1]
```diff
@@ -23,11 +23,11 @@ To see the rendered version of these tags, move this file into the `next/pages/g
...
 
-## Complex Header Where You Definitely Want to Use an Anchor Link {% #complex-header %} 
+## Complex Header Where You Definitely Want to Use an Anchor Link {% #complex-header %}
```
Comment (danielgafni) [inline docs/markdoc-component-documentation.md:26]: sorry for the random trailing whitespace removal in this file, I think it was automatically done by my editor (Zed)

File: docs/next/styles/globals.css [hunk 2]
```diff
@@ -108,11 +108,20 @@ dt > a.reference.internal {
...
+}
+
 .legacy-tag,
 .flag.deprecated {
   @apply inline-flex items-center px-3 py-0.5 rounded-full align-middle text-xs uppercase font-medium bg-yellow-200 text-yellow-700
 }
 
+.legacy-tag,
+.flag.superseded {
+  @apply inline-flex items-center px-3 py-0.5 rounded-full align-middle text-xs uppercase font-medium bg-yellow-200 text-yellow-700
```
[the remark this answers is not in the document]
Comment (salazarm) [reply docs/next/styles/globals.css:122]: no clue how this works tbh. Maybe @hellendag ?

[the remark this answers is not in the document]
Comment (cmpadden) [reply docs/next/styles/globals.css:122]: The docs are on v2 of Tailwind, these are the available colors:

https://v2.tailwindcss.com/docs/customizing-colors

It's possible to customize them to add your own.

[the remark this answers is not in the document]
Comment (cmpadden) [reply docs/next/styles/globals.css:122]: @danielgafni I was able to get orange to work by including it in our theme.

Referencing this guide: https://v2.tailwindcss.com/docs/customizing-colors#disabling-a-default-color

Note that this project is still on Tailwind v2, which has a more limited palette than v3.

```diff
$ git diff next/styles/ next/tailwind.config.js
diff --git a/docs/next/styles/globals.css b/docs/next/styles/globals.css
index e0718b1f7b..3fdd2e7eeb 100644
--- a/docs/next/styles/globals.css
+++ b/docs/next/styles/globals.css
@@ -110,7 +110,7 @@ dt > a.reference.internal {

 .legacy-tag,
 .flag.deprecated {
-  @apply inline-flex items-center px-3 py-0.5 rounded-full align-middle text-xs uppercase font-medium bg-yellow-200 text-yellow-700
+  @apply inline-flex items-center px-3 py-0.5 rounded-full align-middle text-xs uppercase font-medium bg-orange-200 text-orange-700
 }

 .DocSearch--active {
diff --git a/docs/next/tailwind.config.js b/docs/next/tailwind.config.js
index 8dc9e626f2..d200954d7b 100644
--- a/docs/next/tailwind.config.js
+++ b/docs/next/tailwind.config.js
@@ -1,6 +1,7 @@
 /* eslint-disable @typescript-eslint/no-var-requires */

 const defaultTheme = require('tailwindcss/defaultTheme');
+const colors = require('tailwindcss/colors')

 module.exports = {
   important: 'html',
@@ -62,6 +63,7 @@ module.exports = {
           100: '#F5F4F2',
           50: '#FAF9F7',
         },
+        orange: colors.orange,
       },
       animation: {
         wiggle: 'wiggle 1s ease-in-out infinite',
```

**Review by cmpadden — APPROVED** [review]

**Comment by danielgafni** [comment]
### Merge activity

* **Oct 21, 4:08 PM EDT**: A user merged this pull request with [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/25362).

**Pull request merged**